src/library/Cardano/CoinSelection/Algorithm.hs

```diff
@@ -0,0 +1,57 @@
+-- |
+-- Copyright: © 2018-2020 IOHK
+-- License: Apache-2.0
+--
+-- __Submodules__ of this module provide implementations of
+-- __coin selection algorithms__.
+--
+-- Algorithms can be divided into two categories:
+--
+--  * <#generalized-algorithms Generalized Algorithms>
+--
+--        Algorithms that implement the general
+--        'Cardano.CoinSelection.CoinSelectionAlgorithm' interface.
+--
+--  * <#specialized-algorithms Specialized Algorithms>
+--
+--        Algorithms that provide functionality suited to specialized purposes.
+--
+-- = Generalized Algorithms #generalized-algorithms#
+--
+-- The following algorithms implement the general
+-- 'Cardano.CoinSelection.CoinSelectionAlgorithm' interface:
+--
+--  * __"Cardano.CoinSelection.Algorithm.LargestFirst"__
+--
+--        Provides an implementation of the __Largest-First__ algorithm.
+--
+--        When selecting inputs from a given set of UTxO entries, this
+--        algorithm always selects the /largest/ entries /first/.
+--
+--   * __"Cardano.CoinSelection.Algorithm.RandomImprove"__
+--
+--        Provides an implementation of the __Random-Improve__ algorithm.
+--
+--        When selecting inputs from a given set of UTxO entries, this
+--        algorithm always selects entries at /random/.
+--
+--        Once selections have been made, a second phase attempts to /improve/
```

Comment:
you mean `in the next step /improvement/ is attempted` ;-) ?

Reply:
> you mean `in the next step /improvement/ is attempted` ;-) ?

Yes. It could also be written like that. Though writing it with the infinitive form "to improve" means we can link this term with the name of the algorithm in a more obvious way. (Notice that in the description of the algorithm, both "random" and "improve" are emphasized, which together make up the name of the algorithm "random-improve". :smile_cat: )